wagtail/wagtailadmin/static_src/wagtailadmin/js/modal-workflow.js

```diff
@@ -51,9 +51,15 @@ function ModalWorkflow(opts) {
     };
 
     self.loadResponseText = function(responseText) {
-        var response = eval('(' + responseText + ')');
+        var response;
 
-        self.loadBody(response);
+        try {
+            response = JSON.parse( responseText );
+        } catch ( error ) {
+            window.location = window.location.host;
```

Comment:
This seems a bit heavy-handed. Currently, if the modal returns a non-Javascript response, the worst that will happen is that they get a Javascript error or a modal containing a mangled login form or error page - any content they've entered in the 'outer' page will still be salvageable. With this change, they'll be redirected to the site homepage with no warning or explanation, losing their work.